os/board/rtl8721csm/src/component/os/tizenrt/rtk_netmgr.c

```diff
@@ -83,15 +83,86 @@ static wifi_utils_scan_list_s *g_scan_list;
 static int g_scan_num;
 extern struct netdev *ameba_nm_dev_wlan0;
 
+struct ap_scan_list {
+	unsigned int channel;
+	char ssid[WIFI_UTILS_SSID_LEN + 1];
+	unsigned int ssid_length;
+	int rssi;
+	wifi_utils_ap_auth_type_e ap_auth_type;
+	wifi_utils_ap_crypto_type_e ap_crypto_type;
+};
+typedef struct ap_scan_list ap_scan_list_s;
+
+static ap_scan_list_s *saved_scan_list;
+static uint32_t scan_number = 0;
+uint32_t ap_channel;
+
+#include "freertos/wrapper.h"
+#define SCAN_TIMER_DURATION 180000
+_timer scan_timer;
+
+static void _scan_timer_handler(void *FunctionContext)
+{
+	ndbg("scan Timer expired : release saved scan list\r\n");
+	ndbg("scan Timer expired : sizeof(ap_scan_list_s) =%d, scan_number=%d \r\n", sizeof(ap_scan_list_s), scan_number);
+
+	rtw_mfree((unsigned char *)saved_scan_list, sizeof(ap_scan_list_s) * scan_number);
+	scan_number = 0;
+	rtw_del_timer(&(scan_timer));
+}
+
+static int save_scan_list(wifi_utils_scan_list_s *p_scan_list)
+{
+
+	// If application calls scan before scan result free(before timeout), release scan list and cancel timer
+	if (scan_number) {
+		rtw_mfree((unsigned char *)saved_scan_list, sizeof(ap_scan_list_s) * scan_number);
+		rtw_cancel_timer(&(scan_timer));
+		ndbg("scan is called before timeout\r\n");
+	} else {
+		rtw_init_timer(&(scan_timer), &(scan_timer), _scan_timer_handler, &(scan_timer), "dynamic_chk_timer");
+		ndbg("Start scan timer\n");
+	}
+
+	rtw_set_timer(&(scan_timer), SCAN_TIMER_DURATION);
+
+	scan_number	 = 0;
+	saved_scan_list = (ap_scan_list_s *)rtw_zmalloc(sizeof(ap_scan_list_s) * g_scan_num);
+	if (saved_scan_list == NULL)
+		return RTW_NOMEM;

```

Comment:
Is it okay not to stop or cancel the timer?

Reply:
The stop timer "rtw_cancel_timer()" has been used for two cases.
1) The timer is cancelled when it searches for specific AP detail from stored scan result by comparing SSID and set the timer again once if the timer is not deleted -> `wifi_netmgr_utils_connect_ap`
2) If the application calls scan before scan result free(before timeout), release scan list and cancel timer -> `save_scan_list()`